adaptive/learner/averagelearner1D.py

```diff
@@ -0,0 +1,399 @@
+
+        # (no real point right of x) or (no real point left of b)
+        right_loss_is_unknown = (x_right is None) or (not real and x_left is None)
+        if (b is not None) and right_loss_is_unknown:
+            self.losses_combined[x, b] = float("inf")
+
+    def _calc_error_in_mean(self, ys, y_avg, n):
+        variance_in_mean = sum((y - y_avg) ** 2 for y in ys) / (n - 1)
+        t_student = scipy.stats.t.ppf(1 - self.alpha, df=n - 1)
+        return t_student * (variance_in_mean / n) ** 0.5
+
+    def tell_many(self, xs, ys, *, force=False):
+        """The data should be given as:
+                - {x_i: y_i} (only the mean at each point), in which case the
+                  number of samples is assumed to be 1 and the error 0 for all
+                  data points. These points will not be included in
+                  _rescaled_error_in_mean and therefore will not be resampled.
+                - {x_i: [y_i0, y_i1, ...]} (all data samples at each point)."""
+        # If data_samples is given:
+        if isinstance(ys[0], list):
+            self._data_samples.update(zip(xs, ys))
+            ys_avg = np.mean(ys, axis=1)
+            super().tell_many(xs, ys_avg)  # self.data is updated here
+            ys_len = []
+            for x, ys_, y_avg in zip(xs, ys, ys_avg):
+                n = len(ys_)
+                ys_len.append(n)
+                # We include the point in _undersampled_points if there are
+                # less than min_samples samples, disregarding neighbor_sampling.
+                # super().tell_many() sometimes calls self.tell(), which includes
+                # x in _undersampled_points, so we may need to remove it.
+                if n < self.min_samples:
+                    self._undersampled_points.add(x)
+                elif n > self.min_samples:
+                    self._undersampled_points.discard(x)
+                self._error_in_mean[x] = self._calc_error_in_mean(ys_, y_avg, n)
+                self._update_distances(x)
+
+            self._number_samples.update(zip(xs, ys_len))
+
+            for x in xs:
+                if self._number_samples[x] == 1:
+                    self._rescaled_error_in_mean[x] = np.inf
+                elif self._number_samples[x] < self.max_samples:
+                    self._update_rescaled_error_in_mean(x, "resampled")
+            self._data_samples.update(zip(xs, ys))
+
+        # If data is given:
+        else:
+            super().tell_many(xs, ys)  # self.data is updated here
+            self._data_samples.update(zip(xs, ys))
+            self._number_samples.update(zip(xs, [1] * len(xs)))
+            self._error_in_mean.update(zip(xs, [0] * len(xs)))
+            for x in xs:
+                self._update_distances(x)
+
+    def plot(self):
+        """Returns a plot of the evaluated data with error bars.
+
+        Returns
+        -------
+        plot : `holoviews.element.Scatter` (if vdim=1)\
```

Comment:
the return type is incorrect

Reply:
Fixed [here](https://github.com/python-adaptive/adaptive/pull/283/commits/f27cb09b862f9ffcec3b011d95122e0348f94843).